plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/grok/Grok.java

```diff
@@ -0,0 +1,214 @@
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.ingest.processor.grok;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.Object;
+import java.lang.String;
+import java.lang.StringIndexOutOfBoundsException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import com.google.common.base.Charsets;
+import org.jcodings.specific.UTF8Encoding;
+import org.joni.*;
+import org.joni.exception.ValueException;
+
+public class Grok {
+
+    private HashMap<String, String> patternBank;
+    private static final String NAME_GROUP = "name";
+    private static final String SUBNAME_GROUP = "subname";
+    private static final String PATTERN_GROUP = "pattern";
+    private static final String DEFINITION_GROUP = "definition";
+    private static final String GROK_PATTERN =
+            "%\\{" +
+            "(?<name>" +
+            "(?<pattern>[A-z0-9]+)" +
+            "(?::(?<subname>[A-z0-9_:.-]+))?" +
+            ")" +
+            "(?:=(?<definition>" +
+            "(?:" +
+            "(?:[^{}]+|\\.+)+" +
+            ")+" +
+            ")" +
+            ")?" + "\\}";
+    private static final Regex GROK_PATTERN_REGEX = new Regex(GROK_PATTERN.getBytes(Charsets.UTF_8), 0, GROK_PATTERN.getBytes(Charsets.UTF_8).length, Option.NONE, UTF8Encoding.INSTANCE, Syntax.DEFAULT);
+    private Regex compiledExpression;
+    private String expression;
+    private boolean namedCaptures;
+
+
+    public Grok() {
+        this(true);
+    }
+
+    @SuppressWarnings("unchecked")
+    public Grok(boolean namedCaptures) {
+        this.patternBank = new HashMap<>();
+        this.compiledExpression = null;
+        this.expression = null;
+        this.namedCaptures = namedCaptures;
+    }
+
+    /**
+     * loads patterns from a directory path
+     *
+     * @param dir the directory to load the patterns from
+     * @throws IOException when patterns are unable to be read
+     */
+    public void load(Path dir) throws IOException {
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(dir)) {
+            for (Path path : stream) {
+                try (Stream<String> lines = Files.lines(path, Charset.defaultCharset())) {
```

Comment:
`Stream` is unfortunately java8 only, since we are likely to back port ingest to 2.x too we should try to avoid using java8 only constructs and apis. 

But instead of fixing this, I think we can just remove this method as it is not used for now?